File: business/business-oj/src/main/java/com/glowxq/oj/judge/enums/JudgeCompileType.java
package com.glowxq.oj.judge.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum JudgeCompileType implements BaseType {

    /**
     * 编译特判程序
     */
    COMPILE_SPJ("CompileSpj", "编译特判程序"),

    /**
     * 编译交互程序
     */
    COMPILE_INTERACTIVE("CompileInteractive", "编译交互程序");

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static JudgeCompileType matchCode(String code) {
        for (JudgeCompileType sceneType : JudgeCompileType.values()) {
            if (sceneType.getCode().equals(code)) {
                return sceneType;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/judge/JudgeCompileType.ts
import type { BaseEnum } from '@/enums/base';

// 评测编译类型枚举接口
interface JudgeCompileTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 评测编译类型枚举
 * 用于标识不同的编译模式
 */
export const JudgeCompileType: { [key: string]: JudgeCompileTypeEnum } = {
  /**
   * 编译特判程序
   */
  COMPILE_SPJ: {
    code: 'CompileSpj',
    name: '编译SPJ',
    tooltip: '编译特殊判题程序',
    text: '编译特殊判题程序，用于评测特殊判题题目'
  },

  /**
   * 编译交互程序
   */
  COMPILE_INTERACTIVE: {
    code: 'CompileInteractive',
    name: '编译交互式题目',
    tooltip: '编译交互式评测程序',
    text: '编译交互式评测程序，用于评测交互式题目'
  }
};

/**
 * 根据code获取枚举
 * @param code 
 * @returns 枚举值或null
 */
export function matchCode(code: string): JudgeCompileTypeEnum | null {
  return Object.values(JudgeCompileType).find(item => item.code === code) || null;
} 